src/headless/converse-muc.js

```diff
@@ -368,11 +368,17 @@ converse.plugins.add('converse-muc', {
             },
 
             extractReference (text, index) {
-                for (let i=index; i<text.length; i++) {
+                for (let i = index; i < text.length; i++) {
                     if (text[i] !== '@') {
                         continue
                     } else {
-                        const match = text.slice(i+1),
+                        // Hack: Avoid accepting things like url with @ or email
+                        // adresses.
+                        if (i > 1 && text[i - 1] != ' ') {
+                            continue;
+                        }
```

Comment:
The condition for this `if` statement can be made part of the outer condition, like so 
```
if (text[i] !== '@' || (i > 1 && test[i-1] !== ' ')) {
    continue;
} else {
```

Also, as a general rule, use strict equality checking (`===` and `!==`).